### PR TITLE
docs(components):  improve data list docs

### PR DESCRIPTION
**Pull request by darryltec**
## Motivations

Improve tech documentation for DataList

### Added

- Explanation on each compound components

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Review by rodrigoeidelvein — APPROVED** [review]
Nice! Thanks for adding it!

Just a small thing, address it if you think it makes sense, would be to move the BatchActions docs closer to the ItemActions, as they're very similar

And another question is, should we separate the examples? Because currently we have all functionalities in the same story, if we broke them down, maybe it would be easier to understand them

As it's docs, I always love to see feedback from @chris-at-jobber too, wanna do a pass here?

File: docs/components/DataList/DataList.stories.mdx [hunk 1]
```diff
@@ -23,3 +25,201 @@ item.
 If you need to add more information, consider using the
 [DataTable](?path=/story/components-lists-and-tables-datatable-docs--page)
 component instead.
+
+## Implementation guidelines
+
+Since DataList is a large component, we use a mixture of compound components and
+props to separate each part of the component.
+
+## Main wrapper
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:34]:
```suggestion
### Main wrapper
```

File: docs/components/DataList/DataList.stories.mdx [hunk 2]
```diff
@@ -23,3 +25,201 @@ item.
...
+
+DataList is the main component that holds everything and important props.
+
+<ArgsTable of={DataList} />
+
+## Layout
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:40]:
```suggestion
### Layout
```

File: docs/components/DataList/DataList.stories.mdx [hunk 3]
```diff
@@ -23,3 +25,201 @@ item.
...
+        <div>{item.email}</div>
+      </Content>
+    )}
+  </DataList.Layout>
+</DataList>
+```
+
+<ArgsTable of={DataList.Layout} />
+
+## Per item action(s)
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:90]:
```suggestion
### Item actions
```

Maybe debatable, but I think just saying "item actions" maps better to the actual component

File: docs/components/DataList/DataList.stories.mdx [hunk 4]
```diff
@@ -23,3 +25,201 @@ item.
...
+<Tabs>
+  <Tab label="DataList.ItemActions">
+    <ArgsTable of={DataList.ItemActions} />
+  </Tab>
+  <Tab label="DataList.ItemAction">
+    <ArgsTable of={DataList.ItemAction} />
+  </Tab>
+</Tabs>
+
+## Layout actions
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:114]:
```suggestion
### Layout actions
```

File: docs/components/DataList/DataList.stories.mdx [hunk 5]
```diff
@@ -23,3 +25,201 @@ item.
...
+
+```tsx
+<DataList>
+  <DataList.Search onSearch={handleSearch} />
+</DataList>
+```
+
+<ArgsTable of={DataList.Search} />
+
+## Empty state
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:166]:
```suggestion
### Empty state
```

File: docs/components/DataList/DataList.stories.mdx [hunk 6]
```diff
@@ -23,3 +25,201 @@ item.
...
+
+By default, the DataList have a basic empty state message when it is empty or
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:168]:
```suggestion
The DataList has a default empty state message when it is empty or
```

File: docs/components/DataList/DataList.stories.mdx [hunk 7]
```diff
@@ -23,3 +25,201 @@ item.
...
+    type="filtered"
+    message="No results found"
+    action={
+      <Button label="Clear filters" onClick={() => alert("Clear filters")} />
+    }
+  />
+</DataList>
+```
+
+## Batch actions
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:201]:
```suggestion
### Batch actions
```

File: docs/components/DataList/DataList.stories.mdx [hunk 8]
```diff
@@ -23,3 +25,201 @@ item.
...
+
+In order to select multiple items and perform an action on them, you have to set
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:203]:
```suggestion
There are two steps to allow the user to select and perform actions on multiple items:
```

File: docs/components/DataList/DataList.stories.mdx [hunk 9]
```diff
@@ -23,3 +25,201 @@ item.
...
+  onSelectAll={handleSelectAll}
+>
+  <DataList.BatchActions>
+    <DataList.BatchAction label="Send" onClick={handleSend} />
+    <DataList.BatchAction label="Delete" onClick={handleDelete} />
+  </DataList.BatchActions>
+</DataList>
+```
+
+### DataList.BatchAction props
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:223]:
```suggestion
#### DataList.BatchAction props
```

File: docs/components/DataList/DataList.stories.mdx [hunk 10]
```diff
@@ -23,3 +25,201 @@ item.
...
+<DataList>
+  <DataList.Filters>
+    <Combobox />
+    <Button />
+    <Select />
+  </DataList.Filters>
+</DataList>
+```
+
+## Search
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:154]:
```suggestion
### Search
```

File: docs/components/DataList/DataList.stories.mdx [hunk 11]
```diff
@@ -23,3 +25,201 @@ item.
...
+      </>
+    )}
+  </DataList.Layout>
+</DataList>
+```
+
+## Filters
+
+DataList.Filters add filters to the list. This compound component accepts any
+children and drops it above the list of items.
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:142]:
```suggestion
children and inserts it above the list of items.
```

File: docs/components/DataList/DataList.stories.mdx [hunk 12]
```diff
@@ -23,3 +25,201 @@ item.
...
+        <DataList.LayoutActions />
+      </>
+    )}
+  </DataList.Layout>
+</DataList>
+```
+
+## Filters
+
+DataList.Filters add filters to the list. This compound component accepts any
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:141]:
```suggestion
DataList.Filters adds filters to the list. This compound component accepts any
```

File: docs/components/DataList/DataList.stories.mdx [hunk 13]
```diff
@@ -23,3 +25,201 @@ item.
...
+
+        {/* This will place the actions as a part of the layout */}
+        <DataList.LayoutActions />
+      </>
+    )}
+  </DataList.Layout>
+</DataList>
+```
+
+## Filters
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:139]:
```suggestion
### Filters
```

File: docs/components/DataList/DataList.stories.mdx [hunk 14]
```diff
@@ -23,3 +25,201 @@ item.
...
+  </DataList.Layout>
+</DataList>
+```
+
+<ArgsTable of={DataList.Layout} />
+
+## Per item action(s)
+
+DataList.ItemActions allows you to add an action on the item itself such as
+`onClick` or `url`. While the DataList.ItemAction (without an s) child adds more
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:93]: I don't fully understand this guidance

Comment (chris-at-jobber) [reply docs/components/DataList/DataList.stories.mdx:93]: So `ItemActions` is a wrapper for `ItemAction`?

**Review by chris-at-jobber — CHANGES_REQUESTED** [review]
This is great! 90% some hierarchy + grammar suggestions, and one point that I'm not 100% clear on.

**Comment by chris-at-jobber** [comment]
To @rodrigoeidelvein 's point, maybe you could set `### Actions` as a section header and nest Item and Batch underneath?

File: docs/components/DataList/DataList.stories.mdx (hunk 1 above)
Comment (darryltec) [reply docs/components/DataList/DataList.stories.mdx:34]: I was actually doing this so it shows up on the TOC haha

File: docs/components/DataList/DataList.stories.mdx (hunk 14 above)
Comment (darryltec) [reply docs/components/DataList/DataList.stories.mdx:93]: Yup and also item actions also have props that enables the row click to navigate, etc.

I did struggle to write this one haha

File: docs/components/DataList/DataList.stories.mdx (hunk 12 above)
Comment (darryltec) [reply docs/components/DataList/DataList.stories.mdx:141]: grammarly says no? 

image

Comment (chris-at-jobber) [reply docs/components/DataList/DataList.stories.mdx]: Huh I guess that's a weird one...

In the sense that `DataList.Filters` is _one compound component_ that allows you to add filters, then it should be "adds".

In the sense that `Filters` is plural, and you are able to add _multiple filters_, then it should be "add".

File: docs/components/DataList/DataList.stories.mdx (hunk 14 above)
Comment (chris-at-jobber) [reply docs/components/DataList/DataList.stories.mdx:95]: Even looking at the example it's a little hard to follow

It looks like `ItemActions` handles the entire row action, but it's just one action, so the plural name is confusing.

And then it wraps individual `ItemAction` actions

So I would expect a relationship like BottomSheet/BottomSheetOption, but it feels like they're actually more decoupled than that?

Comment (darryltec) [reply docs/components/DataList/DataList.stories.mdx:95]: yeah, a bit more decouple in a sense that you're just telling the data list that here are the list of actions you could do in this item. Also you can fire this onClick if you click the item itself.

I see what you're saying tho that maybe the row click should be its own compound

File: docs/components/DataList/DataList.stories.mdx (hunk 12 above)
Comment (darryltec) [reply docs/components/DataList/DataList.stories.mdx]: you're probably right. grammarly gets confused with new lines and code as english a lot haha

Comment (darryltec) [reply docs/components/DataList/DataList.stories.mdx]: anyway, it's adds now

File: docs/components/DataList/DataList.stories.mdx (hunk 14 above)
Comment (chris-at-jobber) [reply docs/components/DataList/DataList.stories.mdx:95]: OK so maybe this is a clearer way to break it down...

> DataList.ItemActions defines the action that will occur when the user clicks on the list item.
> 
> Provide a DataList.ItemAction for each additional action that you want to add. These additional actions should be presented in a context menu.

Or something like that?

**Review by chris-at-jobber — APPROVED** [review]
